packages/serverpod/lib/src/database/adapters/postgres/database_connection.dart

```diff
@@ -566,12 +566,17 @@ class DatabaseConnection {
     );
 
     return _postgresConnection.runTx<R>(
-      (ctx) {
+      (ctx) async {
         var transaction = _PostgresTransaction(
           ctx,
           session,
         );
-        return transactionFunction(transaction);
+        final result = await transactionFunction(transaction);
+
+        unawaited(
```

Comment:
The async behavior will be a bit tricky in tests, but that should be fine in practice.